server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -79,43 +81,44 @@
-    public static final ParseField POST_FILTER_FIELD = new ParseField("post_filter");
-    public static final ParseField KNN_FIELD = new ParseField("knn");
-    public static final ParseField RANK_FIELD = new ParseField("rank");
-    public static final ParseField MIN_SCORE_FIELD = new ParseField("min_score");
-    public static final ParseField VERSION_FIELD = new ParseField("version");
-    public static final ParseField SEQ_NO_PRIMARY_TERM_FIELD = new ParseField("seq_no_primary_term");
-    public static final ParseField EXPLAIN_FIELD = new ParseField("explain");
-    public static final ParseField _SOURCE_FIELD = new ParseField("_source");
-    public static final ParseField STORED_FIELDS_FIELD = new ParseField("stored_fields");
-    public static final ParseField DOCVALUE_FIELDS_FIELD = new ParseField("docvalue_fields");
-    public static final ParseField FETCH_FIELDS_FIELD = new ParseField("fields");
-    public static final ParseField SCRIPT_FIELDS_FIELD = new ParseField("script_fields");
-    public static final ParseField SCRIPT_FIELD = new ParseField("script");
-    public static final ParseField IGNORE_FAILURE_FIELD = new ParseField("ignore_failure");
-    public static final ParseField SORT_FIELD = new ParseField("sort");
-    public static final ParseField TRACK_SCORES_FIELD = new ParseField("track_scores");
-    public static final ParseField TRACK_TOTAL_HITS_FIELD = new ParseField("track_total_hits");
-    public static final ParseField INDICES_BOOST_FIELD = new ParseField("indices_boost");
-    public static final ParseField AGGREGATIONS_FIELD = new ParseField("aggregations");
-    public static final ParseField AGGS_FIELD = new ParseField("aggs");
-    public static final ParseField HIGHLIGHT_FIELD = new ParseField("highlight");
-    public static final ParseField SUGGEST_FIELD = new ParseField("suggest");
-    public static final ParseField RESCORE_FIELD = new ParseField("rescore");
-    public static final ParseField STATS_FIELD = new ParseField("stats");
-    public static final ParseField EXT_FIELD = new ParseField("ext");
-    public static final ParseField PROFILE_FIELD = new ParseField("profile");
-    public static final ParseField SEARCH_AFTER = new ParseField("search_after");
-    public static final ParseField COLLAPSE = new ParseField("collapse");
-    public static final ParseField SLICE = new ParseField("slice");
-    public static final ParseField POINT_IN_TIME = new ParseField("pit");
-    public static final ParseField RUNTIME_MAPPINGS_FIELD = new ParseField("runtime_mappings");
+    public static final ParseField FROM_FIELD = new ParseField("from"); // base
+    public static final ParseField SIZE_FIELD = new ParseField("size"); // base
+    public static final ParseField TIMEOUT_FIELD = new ParseField("timeout"); // global
+    public static final ParseField TERMINATE_AFTER_FIELD = new ParseField("terminate_after"); // retriever specific
```

Comment:
What's the meaning of the comment "retreiver specific"? It seems that this particular field is used globally for a search request?

Reply:
I intend to remove the comments about these fields once more retrievers are created. I tried to outline the fields for whether or not they would be delegated to the retriever. I admit for `from` and `terminate_after_field` I wasn't completely sure what made sense.

One question I have is what continues to make sense for both global and shard knn? Right now all these parameters work primarily because of the conversion to a `ScoreAndDocQuery` is my understanding. But for a `from` parameter on knn, should that increase k automatically and possibly num_candidates as well?